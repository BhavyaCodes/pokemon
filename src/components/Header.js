import React from "react";
import logo from "../assets/pokemon_logo.png";

function Header({ searchQuery, setSearchQuery }) {
  return (
    <nav className="navbar navbar-expand-md navbar-dark bg-dark">
      <img src={logo} alt="logo" width="180px" height="50px" />

      <button
        type="button"
        className="navbar-toggler"
        data-toggle="collapse"
        data-target="#navbarCollapse"
      >
        <span className="navbar-toggler-icon"></span>
      </button>

      <div className="collapse navbar-collapse" id="navbarCollapse">
        <form className="form-inline ml-auto">
          <input
            type="text"
            className="form-control mr-sm-2"
            placeholder="Search"
            value={searchQuery}
            onChange={(e) => setSearchQuery(e.target.value)}
          />
          <button type="submit" className="btn btn-outline-light">
            Search
          </button>
        </form>
      </div>
    </nav>
  );
}

export default Header;
