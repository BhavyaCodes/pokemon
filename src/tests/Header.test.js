import React from "react";
import { render } from "@testing-library/react";

import Header from "../components/Header";

it("renders Header component without crashing", () => {
  const { asFragment } = render(<Header />);
  expect(asFragment()).toMatchSnapshot();
});
