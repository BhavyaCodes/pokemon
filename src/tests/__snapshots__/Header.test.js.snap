// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Header component without crashing 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-expand-md navbar-dark bg-dark"
  >
    <img
      alt="logo"
      height="50px"
      src="pokemon_logo.png"
      width="180px"
    />
    <button
      class="navbar-toggler"
      data-target="#navbarCollapse"
      data-toggle="collapse"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="collapse navbar-collapse"
      id="navbarCollapse"
    >
      <form
        class="form-inline ml-auto"
      >
        <input
          class="form-control mr-sm-2"
          placeholder="Search"
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-light"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
  </nav>
</DocumentFragment>
`;
